src/main.rs

```diff
@@ -1,12 +1,78 @@
-use std::io::Read;
+use jotdown::{html, Parser};
+use std::{
+    ffi::OsString,
+    fs::{self, File},
+    io::{self, BufWriter},
+    process::exit,
+};
+
+#[derive(Default)]
+struct App {
+    input: Option<OsString>,
+    output: Option<OsString>,
+}
+
+fn parse_args() -> App {
+    let mut app = App::default();
+
+    let mut args = std::env::args_os().skip(1).peekable();
+
+    while let Some(arg) = args.next() {
+        match (arg.to_string_lossy().as_ref(), args.peek()) {
+            ("-h" | "--help", _) => {
+                eprint!("{}", include_str!("./help.txt"));
+                exit(0);
+            }
+            ("-v" | "--version", _) => {
+                eprintln!("jotdown v{}", env!("CARGO_PKG_VERSION"));
+                exit(0);
+            }
+            (flag @ "-o" | flag @ "--output", o) => match o {
+                Some(o) => {
+                    app.output = Some(o.into());
+                    args.next();
+                }
+                None => {
+                    eprintln!("Please supply an argument to {flag}");
+                    exit(1);
+                }
+            },
+            (file, _) if !file.starts_with('-') => app.input = Some(file.into()),
```

Comment:
if you give multiple paths it will use the last one, so for example if you miss the "-o" flag:

```
jotdown in.dj out.html
```
it will silently ignore the in.dj argument, try to use the out.html as input and write to stdout.

Reply:
This is intentional design, as output should be specified with the `-o` flag. I thought it would be simpler to just ignore multiple arguments and go with the last one, but perhaps we should disallow multiple arguments?

Reply:
I think if the usage is
```
usage: jotdown [option] [file]
```
instead of
```
usage: jotdown [option] [file] ...
```
I would expect it to only accept zero or one positional arguments.

For optional arguments, on the other hand, it is quite common to allow overwriting so one can just append it to the end of some unknown existing arguments.

On a related note, I realized that pandoc, for example, does accept multiple files and it actually concatenates them. So perhaps there is a usecase for this?

Because of this, I always get confused when forgetting the -o flag to pandoc. : p

Reply:
Then let's go for overwriting the optional `file` argument and rather update the help text accordingly?

As far as concatenation goes, I'm not sure it would be that well-defined. Since newlines matter, how many should we insert between files? I think it's better to restrict to one file argument, and if users need multiple files concatenated, they can use pipes (`cat file1.dj file2.dj | jotdown`.)

Reply:
> Then let's go for overwriting the optional `file` argument and rather update the help text accordingly?

Heh, typo/confusion by me, by optional I was thinking of optional _flags_, e.g. the `-o output_file` argument.

Rather, I would probably prefer an error when providing too many _positional_ arguments, something like:

```rs
eprint!("too many arguments\n\n{}", include_str!("./help.txt"));
```

> As far as concatenation goes, I'm not sure it would be that well-defined. Since newlines matter, how many should we insert between files? I think it's better to restrict to one file argument, and if users need multiple files concatenated, they can use pipes (`cat file1.dj file2.dj | jotdown`.)

I would guess one newline might be reasonable, but no need to implement anything now. It can be done later if we encounter a usecase.

Reply:
Fixed!